tensorflow/core/kernels/dml_resource_scatter_ops.cc

```diff
@@ -421,6 +421,10 @@ class DmlResourceScatterNDBinaryKernel : public DmlKernel {
     auto indices = dml::InputTensor(graph, 1, inputs[1]);
     auto updates = dml::InputTensor(graph, 2, inputs[2]);
     auto strides = dml::InputTensor(graph, 3, inputs[3]);
+
+    // TODO: Remove the Is64BitIntegerType hack when DML has a more solid
+    // solution for 64 bit datatypes
+    // TFDML #24881131
     auto result = BinaryOp()(graph, input, indices, updates, strides,
                              Is64BitIntegerType(indices_dtype));
 

```

Comment:
Does scatter already work with the `result = dml::ConvertInt32ToInt64(scope, result);` that you added elsewhere?

Reply:
Scatter isn't registered for int64 output since CUDA doesn't register it either. The Is64BitIntegerType here is for the input indices.